### PR TITLE
Feature: Refactor American/British Translator Project Boilerplate for a Backend Focus to Improve Testability

### PR DESCRIPTION
**Pull request by SaintPeter**
As discussed here:
https://github.com/freeCodeCamp/freeCodeCamp/issues/39692

@nhcarrigan and I refactored this project to have a more backend focus to improve testability by the fCC test infrastructure.

Key Changes:
* Removed all front-end related User Stories 
* Rewrote the front-end to provide minimal features, treating it as a basic API endpoint test
* Moved all logic to the back-end.
* Added unit tests to cover highlighting
* Added functional tests to cover the new API endpoint

NOTE: This change should not be merged until the (as yet uncreated) `/learn` PR has been completed.

### CODE REVIEW

**Comment by SaintPeter** [comment]
Removed user stories as described in https://github.com/freeCodeCamp/freeCodeCamp/issues/39721

**Review by moT01 — APPROVED** [review]
As far as I can tell, I think this boilerplate LGTM 🎉

I made some suggestions on the other two PR's.

File: tests/1_unit-tests.js [hunk 1]
```diff
@@ -9,171 +9,188 @@
...
-      });
+      // done();
+    });
 
-      test("I've just got bits and bobs in my bum bag. --> I've just got odds and ends in my fanny pack.", done => {
-        const input = "I've just got bits and bobs in my bum bag.";
-        const output = "I've just got odds and ends in my fanny pack.";
+    test('Tea time is usually around 4 or 4.30. --> Tea time is usually around 4 or 4:30.', done => {
+      const input = 'Lunch is at 12:15 today.';
+      const output = 'Lunch is at 12.15 today.';
```
Comment (RandellDawson) [inline tests/1_unit-tests.js]: Shouldn't this be either:
```js
    test('Tea time is usually around 4 or 4.30. --> Tea time is usually around 4:00 or 4:30.', done => {
      const input = 'Tea time is usually around 4 or 4.30 today.';
      const output = 'Tea time is usually around 4:00 or 4:30 today.';
```
OR
```js
    test('Tea time is usually around 4 or 4.30. --> Tea time is usually around 4or 4:30.', done => {
      const input = 'Tea time is usually around 4 or 4.30 today.';
      const output = 'Tea time is usually around 4 or 4:30 today.';
```

File: tests/1_unit-tests.js [hunk 2]
```diff
@@ -9,171 +9,188 @@
...
+      // done();
+    });
+
+  });
+
+  suite('Highlight Translated Text', () => { 
+    suite('Highlight American to British English', () => {
+      test('Highlight "Mangoes" in the "Mangoes are my favorite fruit." translation', done => {
+        const input = 'Mangoes are my favorite fruit.';
+        const output = '<span class="highlight">Mangoes</span> are my favourite fruit.';
```
Comment (RandellDawson) [inline tests/1_unit-tests.js]: Shouldn't this be:
```suggestion
        const output = 'Mangoes are my <span class="highlight">favourite</span> fruit.';
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@SaintPeter I have a question and a suggestion.  If you make any of these changes, will you have to change anything else to avoid test failures?

File: tests/1_unit-tests.js (hunk 1 above)
Comment (naomi-lgbt) [reply tests/1_unit-tests.js]: Yes.

File: tests/1_unit-tests.js (hunk 2 above)
Comment (naomi-lgbt) [reply tests/1_unit-tests.js]: Also yes.

**Comment by naomi-lgbt** [comment]
@SaintPeter did we lose a commit here? I feel like these are things we already fixed? (Or am I just losing it?)

**Comment by SaintPeter** [comment]
I have fixed these issues here and in the demo project.

File: routes/api.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+*
+*/
+
+'use strict';
+
+const Translator = require('../components/translator.js');
+
+module.exports = function (app) {
+  
+  let translator = new Translator();
```
Comment (RandellDawson) [inline routes/api.js]: I notice this line here, but in the PR you created for the example project, you did not actually incorporate this line as you never instantiated `Translator`.  I suppose I am fine leaving it, but I would prefer using `const`.
```suggestion
  const translator = new Translator();
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@SaintPeter I have one small question/suggestion.  Everything else seems fine.  I tested it out but simply copying/pasting the relevant code from the example project you created and then tested against the corresponding `/learn` PR and everything passes.

Great job!

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
OK, this PR looks good now.

@moT01 I have approved https://github.com/freeCodeCamp/freeCodeCamp/pull/39822 and https://github.com/freeCodeCamp/demo-projects/pull/23, so if you approve, we can merge this as soon as the `/learn` PR hits production.

**Review by moT01 — APPROVED** [review]
👍 @SaintPeter @nhcarrigan 🚀 The main PR is now on production so this can get merged 😄

**Pull request merged**